subworkflows/nf-core/merge_index_bam.nf

```diff
@@ -15,19 +15,27 @@ workflow MERGE_INDEX_BAM {
     ch_versions = Channel.empty()
 
     // Figuring out if there is one or more bam(s) from the same sample
-    bam.branch{
-        single:   it[1].size() == 1
-        multiple: it[1].size() > 1
+    bam.groupTuple().branch{ meta, bam ->
+        single:   bam.size() == 1
+            [meta, bam.transpose()[0]]
```

Comment:
can you add a comment what happens here?

Reply:
actually I'm still not sure what is happening there... which is why I pinged @drpatelh and @mahesh-panchal on Slack, hoping they would come check this PR

Reply:
```suggestion
            [meta, *bam]
```